File: 1915-check-if-one-string-swap-can-make-strings-equal/check-if-one-string-swap-can-make-strings-equal.py
class Solution:
    def areAlmostEqual(self, s1: str, s2: str) -> bool:
        if s1 == s2:
            return True  # Strings are already equal
        
        idx = -1
        s1_list = list(s1)  # Convert s1 to a list for mutability
        for i in range(len(s1)):
            if s1[i] != s2[i]:
                if idx == -1:
                    idx = i  # Store the first mismatch index
                else:
                    # Swap characters in the list
                    s1_list[i], s1_list[idx] = s1_list[idx], s1_list[i]
                    # Compare the modified list with s2
                    return ''.join(s1_list) == s2
        
        # If only one mismatch, return False (since no swap can fix it)
        return False

File: 1915-check-if-one-string-swap-can-make-strings-equal/check-if-one-string-swap-can-make-strings-equal.java
class Solution {
    public boolean areAlmostEqual(String s1, String s2) {
        if(s1.equals(s2)) return true ;
        char[] s = s1.toCharArray();
        int idx = -1 ;
        for(int i = 0 ; i < s.length ; i ++){
            if(s[i] != s2.charAt(i)){
                if(idx == -1){
                    idx = i ;
                }else{
                    char temp = s[i];
                    s[i] = s[idx];
                    s[idx] = temp;
                    return (new String(s)).equals(s2);
                }
            }
        }
        return false ;
    }
}